Toast: Prevent error being thrown if opened without returnElements setting

brentswisher opened the pull request:
**This change:** (check at least one)

- [ ] Adds a new feature
- [X] Fixes a bug
- [ ] Improves maintainability
- [ ] Improves documentation
- [ ] Is a release activity

**Is this a breaking change?** (check one)

- [ ] Yes
- [X] No

**Is the:** (complete all)

- [X] Title of this pull request clear, concise, and indicative of the issue number it addresses, if any?
- [X] Test suite(s) passing?
- [X] Code coverage maximal?
- [X] Changeset added?
- [ ] Component status page up to date?

**What does this change address?**
Fixes #571 by preventing the default toaster `returnElements` value from being overwritten to `undefined` if a toast event is created without specifying it in the `pharos-toast-open` event.

**How does this change work?**
If the `returnElements` coming from the event details is `undefined` it will be defaulted to an empty array

daneah reviewed (commented):
Is there a test that could exercise the bug?

brentswisher commented:
>Is there a test that could exercise the bug?

I updated the existing mock to not pass an explicit `returnElements` in, before it was passing `[]` So, that is now testing the bug without adding an explicit test. 

I changed the test first, saw it failing, made my change, saw it passing so I do think it is covering the changes.  

Not opposed to adding one though, if you think that would be better?

daneah commented:
> Not opposed to adding one though, if you think that would be better?

That seems sufficient!
